File: php/features.php
<?php
declare(strict_types=1);

// GlaxWeather SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GlaxWeatherFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GlaxWeatherBaseFeature();
            case "test":
                return new GlaxWeatherTestFeature();
            default:
                return new GlaxWeatherBaseFeature();
        }
    }
}

File: py/features.py
# GlaxWeather SDK feature factory

from feature.base_feature import GlaxWeatherBaseFeature
from feature.test_feature import GlaxWeatherTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GlaxWeatherBaseFeature(),
        "test": lambda: GlaxWeatherTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
